src/firmware_variables/variables.py

```diff
@@ -86,3 +89,65 @@ def delete_variable(name, *args, **kwargs):
     :param attributes: @see Attributes
     """
     set_variable(name, value=b"", *args, **kwargs)
+
+
+class NTError(OSError):
+    def __init__(self, nt_status):
+        self.nt_status = nt_status
+
+    def __str__(self):
+        return "[Error 0x{:08X}]".format(self.nt_status)
+
+
+def get_all_variables_names():
+    """
+    Get the names of all the UEFI Variables in the system.
+    :return: A list of tuples containing namespace and variable name.
+    """
+    INFORMATION_VARIABLE_NAMES = 1
+    STATUS_SUCCESS = 0
+    STATUS_BUFFER_TOO_SMALL = 0xc0000023
+    SIZEOF_GUID = 16
+    SIZEOF_DWORD = 4
+
+    def parse_variable_entry(entry_data):
+        guid = uuid.UUID(bytes_le=entry_data[:SIZEOF_GUID])
+        name = entry_data[SIZEOF_DWORD:]
+        return "{{{}}}".format(str(guid)), name
+
+    def parse_firmware_variables_buffer(raw_buf):
+        variables = []
+        current_offset = 0
+        while True:
+            next_offset, = struct.unpack('<I', raw_buf[current_offset: current_offset + SIZEOF_DWORD])
+            if next_offset == 0:
+                break
+
+            entry_data = raw_buf[current_offset + SIZEOF_DWORD: current_offset + next_offset]
+            variables.append(parse_variable_entry(entry_data))
+            current_offset += next_offset
+
+        return variables
+
+    verify_uefi_firmware()
+
+    buf = create_string_buffer(0)
+    length = DWORD(0)
+    status = nt_enumerate_system_firmware_values_ex(
+        INFORMATION_VARIABLE_NAMES,
+        buf,
+        pointer(length)
+    )
+    if status != STATUS_BUFFER_TOO_SMALL:
+        raise NTError(status)
+
+    buf = create_string_buffer(length.value)
```

Comment:
do this in while because it has TOCTOU issue - someone can add new variable between the two functions and then it will fail